relay-server/src/actors/project_buffer.rs

```diff
@@ -407,6 +400,12 @@ impl BufferService {
     ) -> Result<(), QueueKey> {
         loop {
             // By default this creates a prepared statement which is cached and re-used.
+            //
+            // Removing envelopes from the on-disk buffer in batches has following implicatations:
```

Comment:
nit: I'm not sure if I like the "on-disk" here since this in the end is an abstraction, but let's move forward with this for now.